### PR TITLE
Switch to CentOS stream9 builder

### PR DESCRIPTION
**Pull request by maya-r**
**What this PR does / why we need it**:
Had some trouble building the centos builder on CI.
@mhenriks reverted it to be able to make some changes - let's re-introduce this change.

**Special notes for your reviewer**:

**Release note**:

```release-note
Switch to CentOS Stream9 for building images
```

### CODE REVIEW

**Comment by maya-r** [comment]
/hold
needs https://github.com/kubevirt/project-infra/pull/1829 first.

File: hack/build/docker/builder/Dockerfile [hunk 1]
```diff
@@ -30,35 +24,37 @@ RUN sed -i 's/proxy = None//gI' /tmp/fedora_ci.dnf.repo && \
...
+    source /etc/profile.d/gimme.sh && \
+    eval $(go env) && \
+    go get github.com/onsi/ginkgo/ginkgo && \
+    go get golang.org/x/tools/cmd/goimports && \
+    go get mvdan.cc/sh/cmd/shfmt && \
+    go get github.com/mattn/goveralls && \
+    go get -u golang.org/x/lint/golint && \
+    go get github.com/rmohr/go-swagger-utils/swagger-doc && \
+    go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 && \
+    go get github.com/securego/gosec/v2/cmd/gosec@0ce48a5 && \
```
Comment (mhenriks) [inline hack/build/docker/builder/Dockerfile:56]: Why going back to `go get` Isn't it deprecated?

Comment (maya-r) [reply hack/build/docker/builder/Dockerfile:56]: Didn't notice this is a change I should adopt, will change.

**Comment by maya-r** [comment]
/hold cancel
bootstrap image was updated.

**Comment by awels** [comment]
So #1983 is for updating the output containers to use centos stream 9? I am not sure if we can separate the two. The builder image builds the binaries which eventually end up in the target containers. So we would have binaries build on centos 9 being deposited into fedora 33 containers. I am fine with a short period of this being out of sync, but is #1983 ready as well?

**Comment by alicefr** [comment]
If you don't switch at the same time the builder and the base container image, you could end up having a mismatch with the dynamic libraries and symbols. It already happen with a too old fedora package for libnbd (see [this comment](https://github.com/kubevirt/containerized-data-importer/pull/2081#issuecomment-1007242976)). This could happen even more likely if the 2 environments have 2 different OSes.

**Comment by maya-r** [comment]
Yeah, unfortunately our current workflow requires there to be a small window of mismatch between builder and base image.
We only update the builder after the push is done, and cannot use it for the PR we are currently building.

**Comment by maya-r** [comment]
@awels both PRs are as ready as they can be - the second one needs to update BUILDER_IMAGE to the checksum that will be generated once this PR is merged.

**Pull request merged**